### PR TITLE
Add TextShow instance for Aeson KeyMap

### PR DESCRIPTION
**Pull request by flip111**
https://github.com/RyanGlScott/text-show-instances/issues/17

### CODE REVIEW

File: src/TextShow/Data/Aeson.hs [hunk 1]
```diff
@@ -31,3 +31,8 @@ instance TextShow Value where
         $ fromText "Object (fromList "
         <> showbPrec 11 (KM.toAscList xs)
         <> singleton ')'
+
+instance TextShow v => TextShow (KM.KeyMap v) where
+    showbPrec d xs = showbParen (d > 10)
+        $ fromText "fromList "
+        <> showbPrec 11 (KM.toAscList xs)
```
Comment (RyanGlScott) [inline src/TextShow/Data/Aeson.hs]: Can you add a `TextShow1 KM.KeyMap` instance as well?

**Review by RyanGlScott — CHANGES_REQUESTED** [review]
Please also add corresponding tests in [`Spec.Data.AesonSpec`](https://github.com/RyanGlScott/text-show-instances/blob/47a786b7379298bcbc5d99ee883f30a1f9d205e5/tests/Spec/Data/AesonSpec.hs#L25-L30). Thanks!

File: src/TextShow/Data/Aeson.hs (hunk 1 above)
Comment (flip111) [reply src/TextShow/Data/Aeson.hs]: I'm not sure how to make this instance. There is no `liftShowbPrec2` defined for `KeyMap` (not yet). And `showbUnaryListWith` doesn't seem to apply (at least i think so)

Comment (RyanGlScott) [reply src/TextShow/Data/Aeson.hs]: Something like this should suffice:

```hs
import TextShow (TextShow1(..), ...)
import TextShow.Utils (showbUnaryListWith)

instance TextShow v => TextShow (KM.KeyMap v) where
    showbPrec = showbPrec1

instance TextShow1 KM.KeyMap where
    liftShowbPrec sp sl d xs =
      showbUnaryListWith (liftShowbList sp sl) d (KM.toAscList xs)
```

**Review by RyanGlScott — APPROVED** [review]

**Comment by flip111** [comment]
Thank you for merging !

**Pull request merged**